newIDE/app/src/InstancesEditor/TileMapPaintingPreview.js

```diff
@@ -339,42 +349,46 @@ class TileMapPaintingPreview {
         sceneToTileMapTransformation: this.sceneToTileMapTransformation,
       }
     );
+    if (spritesCoordinatesInTileMapGrid.length === 0) {
+      console.warn("Could't get coordinates to render in tile map grid.");
+      return;
+    }
 
     const workingPoint = [0, 0];
 
-    spritesCoordinatesInTileMapGrid.forEach(({ x, y }) => {
-      let sprite;
+    const sprite = new PIXI.TilingSprite(texture);
+
+    sprite.tileScale.x =
+      (tileMapTileSelection.flipHorizontally ? -1 : +1) *
+      this.viewPosition.toCanvasScale(scaleX);
+    sprite.tileScale.y =
+      (tileMapTileSelection.flipVertically ? -1 : +1) *
+      this.viewPosition.toCanvasScale(scaleY);
+    sprite.width = spriteWidth;
+    sprite.height = spriteHeight;
+
+    const allXCoordinates = spritesCoordinatesInTileMapGrid.map(({ x }) => x);
+    const allYCoordinates = spritesCoordinatesInTileMapGrid.map(({ y }) => y);
+    const minX = Math.min(...allXCoordinates);
+    const maxX = Math.max(...allXCoordinates);
+    const minY = Math.min(...allYCoordinates);
+    const maxY = Math.max(...allYCoordinates);
```

Comment:
While it looks elegant, probably a better idea to just do a for loop on spritesCoordinatesInTileMapGrid and a `if (x < minX) minX = x;`, because otherwise you're allocating 2 arrays of easily 500-200 elements at each render (not bad, but ok) and calling Math.min/max with 500-2000 parameters - JS engines are certainly not happy with this.

A good old for loop while be easily optimized by JS engines and even CPUs.

Reply:
Yeah good idea, it was a bit brainless 😅